consensus/istanbul/backend/announce.go

```diff
@@ -1376,22 +1356,22 @@ func (sb *Backend) SetEnodeCertificateMsgMap(enodeCertMsgMap map[enode.ID]*istan
 		}
 	}
 
-	sb.enodeCertificateMsgMapMu.Lock()
-	defer sb.enodeCertificateMsgMapMu.Unlock()
+	m.enodeCertificateMsgMapMu.Lock()
+	defer m.enodeCertificateMsgMapMu.Unlock()
 
 	// Already have a more recent enodeCertificate
-	if *enodeCertVersion < sb.enodeCertificateMsgVersion {
-		logger.Error("Ignoring enode certificate msgs since it's an older version", "enodeCertVersion", *enodeCertVersion, "sb.enodeCertificateMsgVersion", sb.enodeCertificateMsgVersion)
+	if *enodeCertVersion < m.enodeCertificateMsgVersion {
+		logger.Error("Ignoring enode certificate msgs since it's an older version", "enodeCertVersion", *enodeCertVersion, "sb.enodeCertificateMsgVersion", m.enodeCertificateMsgVersion)
 		return istanbul.ErrInvalidEnodeCertMsgMapOldVersion
-	} else if *enodeCertVersion == sb.enodeCertificateMsgVersion {
+	} else if *enodeCertVersion == m.enodeCertificateMsgVersion {
 		// This function may be called with the same enode certificate.
 		// Proxied validators will periodically send the same enode certificate to it's proxies,
 		// to ensure that the proxies to eventually get their enode certificates.
 		logger.Trace("Attempting to set an enode certificate with the same version as the previous set enode certificate's")
 	} else {
 		logger.Debug("Setting enode certificate", "version", *enodeCertVersion)
-		sb.enodeCertificateMsgMap = enodeCertMsgMap
-		sb.enodeCertificateMsgVersion = *enodeCertVersion
+		m.enodeCertificateMsgMap = enodeCertMsgMap
+		m.enodeCertificateMsgVersion = *enodeCertVersion
```

Comment:
the two functions below should be part of the announceManager if we move ownership of the valEnodeTable to it

Reply:
Yes it's something I'll definitely explore in future PRs. Moving the valEnodeTable so soon seemed like a lot.